### PR TITLE
TRUNK-4869: added deathDate validation if it is set in future. BirthD…

### PR DESCRIPTION
**Pull request by samshuster**
## Related Issue

see https://issues.openmrs.org/browse/TRUNK-

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

…ate validation already existed

TRUNK-4869: added javadocs to all methods. Fixed naming

### CODE REVIEW

**Comment by samshuster** [comment]
first pull request for this project. I think i followed all directions!

File: api/src/test/java/org/openmrs/validator/PersonValidatorTest.java [hunk 1]
```diff
@@ -166,8 +185,7 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() throws Ex
 		person.setBirthdate(new Date());
 		person.setGender("too long text too long too long text too long text  too long text");
 		person.setPersonVoided(true);
-		person
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/validator/PersonValidatorTest.java]: Did you touch this? Or did you forget to run mvn clean install before committing as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

Comment (dkayiwa) [inline api/src/test/java/org/openmrs/validator/PersonValidatorTest.java]: We only recommend doing mvn clean install on the whole project. This should not have any formatting in places you have not touched.

**Comment by samshuster** [comment]
I see thanks for your patience @dkayiwa. I removed the formatting change that was introduced by accident.

I made sure the project builds with a mvn clean install.

**Pull request merged**